### PR TITLE
feat(veeam-backup): add unit tests + small fixes

### PR DESCRIPTION
**Pull request by chipp972**
ref: MANAGER-15276

| Question         | Answer
| ---------------- | ---
| Branch?          | `develop` 
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-15276 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

### CODE REVIEW

File: packages/manager-react-components/src/hooks/iam/useOvhIam.tsx
Comment (darsene) [inline packages/manager-react-components/src/hooks/iam/useOvhIam.tsx:1]: All changes on manager-react-components, should be backported in another PR

**Review by pauldkn — COMMENTED** [review]
Except for manager-react-components changes, ✅

File: packages/manager-react-components/src/hooks/iam/useOvhIam.tsx
Comment (chipp972) [reply packages/manager-react-components/src/hooks/iam/useOvhIam.tsx:1]: Here are the manager-components PRs for those changes

https://github.com/ovh/manager/pull/13175
https://github.com/ovh/manager/pull/13169

File: packages/manager/apps/veeam-backup/tests/dashboard.spec.tsx [hunk 1]
```diff
@@ -1,21 +1,70 @@
...
-  it('displays the onboarding page if there is 0 backup', async () => {
-    await setupTest({ nbBackup: 0 });
-    expect(screen.getByText(labels.onboarding.description)).toBeVisible();
-    expect(
-      screen.queryByText(labels.listing.description),
-    ).not.toBeInTheDocument();
+  it('displays the dashboard page when clicking on the link', async () => {
+    await setupTest();
+    await goToDashboard(backupList[0].iam.displayName);
   });
```
[the remark this answers is not in the document]
Comment (chipp972) [reply packages/manager/apps/veeam-backup/tests/dashboard.spec.tsx:22]: Actually the goToDashboard function contains an expect that checks if some text specific to the dashboard is displayed

I put it inside a function to be able to reuse it across the other dashboard tests and I create a separate test for this to assess if any problem occurs, if it is the action to go to the dashboard that fails or if it is the actions after that fails

**Review by darsene — APPROVED** [review]

File: packages/manager/apps/veeam-backup/src/pages/listing/DatagridCell.component.tsx [hunk 2]
```diff
@@ -115,17 +115,11 @@ export const RegionCell = ({
...
-    ?.location;
-
-  if (isLoading) {
-    return <OsdsSkeleton />;
-  }
-  return isDashboard ? (
-    <Description>{label}</Description>
-  ) : (
-    <DataGridTextCell>{label}</DataGridTextCell>
+  const Wrapper = isDashboard ? Description : DataGridTextCell;
```
Comment (rjamet-ovh) [inline packages/manager/apps/veeam-backup/src/pages/listing/DatagridCell.component.tsx:118]: you could remove the the dashboard use-case as this component is intended as a Datagrid cell component. The common part is already handled by the `<Region />` component anyway

**Review by rjamet-ovh — APPROVED** [review]
